src/gc.c

```diff
@@ -833,6 +833,7 @@ void jl_gc_track_malloced_array(jl_ptls_t ptls, jl_array_t *a)
         ptls->heap.mafreelist = ma->next;
     }
     ma->a = a;
+    ma->how = how;
```

Comment:
Since now you take over the commit I feel like it should be reasonable to request this to use a tagged bit instead.